File: Python/804_Unique_Morse_Code_Words.py
class Solution:
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        MORSE = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        
        return len({"".join(MORSE[ord(c) - ord("a")] for c in word) for word in words})

File: Go/804_unique_morse_code_words.go
func uniqueMorseRepresentations(words []string) int {
    uniq := make(map[string]struct{})
    MORSE := []string{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."}
    
    for _, word := range words {
        morseWord := ""
        for _, c := range word {
            morseWord += MORSE[c - 'a']
        }
        
        if _, present := uniq[morseWord]; !present {
            uniq[morseWord] = struct{}{}
        }
    }
    return len(uniq)
}
